src/Compilers/CSharp/Portable/BoundTree/BoundDecisionDag.cs

```diff
@@ -98,88 +93,104 @@ public ImmutableArray<BoundDecisionDagNode> TopologicallySortedNodes
             // a node's successors before the node, the replacement should always be in the cache when we need it.
             var replacement = PooledDictionary<BoundDecisionDagNode, BoundDecisionDagNode>.GetInstance();
 
+            Func<BoundDecisionDagNode, BoundDecisionDagNode> getReplacementForChild = n => replacement[n];
+
             // Loop backwards through the topologically sorted nodes to translate them, so that we always visit a node after its successors
             for (int i = sortedNodes.Length - 1; i >= 0; i--)
             {
                 BoundDecisionDagNode node = sortedNodes[i];
                 Debug.Assert(!replacement.ContainsKey(node));
-                BoundDecisionDagNode newNode = makeReplacement(node);
+                BoundDecisionDagNode newNode = makeReplacement(node, getReplacementForChild);
                 replacement.Add(node, newNode);
             }
 
             // Return the computed replacement root node
             var newRoot = replacement[this.RootNode];
             replacement.Free();
             return this.Update(newRoot);
+        }
+
+        /// <summary>
+        /// A trivial node replacement function for use with <see cref="Rewrite(Func{BoundDecisionDagNode, Func{BoundDecisionDagNode, BoundDecisionDagNode}, BoundDecisionDagNode})"/>.
+        /// </summary>
+        /// <param name="dag"></param>
+        /// <param name="replacement"></param>
+        /// <returns></returns>
+        public BoundDecisionDagNode TrivialReplacement(BoundDecisionDagNode dag, Func<BoundDecisionDagNode, BoundDecisionDagNode> replacement)
```

Comment:
>public [](start = 8, length = 6)

`static`?

Reply:
It is intended to be used as a fallback by outside callers of `Rewrite` as a fallback. There is one caller in this iteration, but an outside caller in a later iteration where we rewrite for tuple literal inputs to a switch.

---
In reply to: [180570098](https://github.com/dotnet/roslyn/pull/25963#discussion_r180570098) [](ancestors = 180570098)